cli/src/pcluster/models/cluster_config.py

```diff
@@ -695,15 +701,39 @@ def vcpus(self):
         )
 
     @property
-    def pass_cpu_options_in_launch_template(self):
+    def pass_cpu_options_in_launch_template(self) -> bool:
         """Check whether CPU Options must be passed in launch template for head node."""
-        instance_type_info = AWSApi.instance().ec2.get_instance_type_info(self.instance_type)
-        return self.disable_simultaneous_multithreading and instance_type_info.is_cpu_options_supported_in_lt()
+        return self.disable_simultaneous_multithreading and self._instance_type_info.is_cpu_options_supported_in_lt()
+
+    @property
+    def is_ebs_optimized(self) -> bool:
+        """Return True if the instance has optimized EBS support."""
+        return self._instance_type_info.is_ebs_optimized()
 
     @property
-    def instance_type_info(self):
-        """Return head node instance type information as returned from aws ec2 describe-instamce-types."""
-        return AWSApi.instance().ec2.get_instance_type_info(self.instance_type)
+    def max_network_interface_count(self) -> int:
+        """Return max number of NICs for the instance."""
+        return self._instance_type_info.max_network_interface_count()
+
+    @property
+    def _instance_type_info(self) -> InstanceTypeInfo:
+        """Return head node instance type information as returned from aws ec2 describe-instance-types."""
+        if not self.__instance_type_info:
+            self.__instance_type_info = AWSApi.instance().ec2.get_instance_type_info(self.instance_type)
+        return self.__instance_type_info
```

Comment:
Why store the info in an internal attribute. Shall we do all the caching in AWSApi layer?

Reply:
Good point, yes I know it's cached on aws api layer but given that I'm using this a lot of times for a lot of properties I want to be sure we're caching this value even if we remove the cache from aws api layer.